lxd/cluster/heartbeat.go

```diff
@@ -61,16 +72,16 @@ func (hbState *APIHeartbeat) Update(fullStateList bool, raftNodes []db.RaftNode,
 
 	// Add raft nodes first with the raft flag set to true, but missing LastHeartbeat time.
 	for _, node := range raftNodes {
-		member, exists := hbState.Members[node.ID]
+		member, exists := hbState.memberByAddress(node.Address)
 		if !exists {
 			member = APIHeartbeatMember{
-				ID:      node.ID,
 				Address: node.Address,
 			}
 		}
 
+		member.RaftID = node.ID
 		member.Raft = true
-		hbState.Members[node.ID] = member
+		hbState.Members[member.ID] = member
```

Comment:
@stgraber on the first pass this will still be the raft node ID, meaning on the 2nd pass below the allNodes loop, it wont match by ID and wont add heartbeat time correctly.

Reply:
@stgraber sorry to be clearer, on the first pass this means the raft nodes added will have a member ID of zero.

Reply:
Ah yeah, indeed, that's a problem. I guess we should build a temporary map that stores things by address and then convert that back into the APIHeartbeat format at the end.